api/restapi/restapi.go

```diff
@@ -70,33 +75,26 @@ type peerAddBody struct {
 // NewRESTAPI creates a new REST API component. It receives
 // the multiaddress on which the API listens.
 func NewRESTAPI(apiMAddr ma.Multiaddr) (*RESTAPI, error) {
-	n, addr, err := manet.DialArgs(apiMAddr)
-	if err != nil {
-		return nil, err
-	}
-	l, err := net.Listen(n, addr)
-	if err != nil {
-		return nil, err
-	}
-	return newRESTAPI(apiMAddr, l)
+	return NewRESTAPIWithConfig(apiMAddr, &Config{})
 }
 
-// NewTlsRESTAPI creates a new REST API component that uses TLS for security
-// (authentication, encryption). It receives the multiaddress on which the API
-// listens, as well as paths to certificate and private key files
-func NewTLSRESTAPI(apiMAddr ma.Multiaddr, tlsCfg *tls.Config) (*RESTAPI, error) {
+// NewRESTAPI creates a new REST API component. It receives
+// the multiaddress on which the API listens.
+func NewRESTAPIWithConfig(apiMAddr ma.Multiaddr, cfg *Config) (*RESTAPI, error) {
```

Comment:
Should include the apiMaddr as part of the configuration I guess

Reply:
Sounds good -- the only difference that I see so far is that the `apiMAddr` is require, while everything in the `cfg` is optional. Not sure if that difference matters much, but for now I'm combining them into a single config (so we'll have `NewRESTAPI(cfg *Config)`).